RELEASE.md

```diff
@@ -34,4 +36,39 @@ resolvers ++= Seq(
   )
 ```
 
-## TODO(tomerk): Add details about publishing a full release
+## Full Release
+
+1. Update the build.sbt and the Makefile to the new version
+
+2. Cut a release branch for major version releases
```

Comment:
Write it from the perspective of cutting a brand new release from the `master` branch. We can add a subsection below for incremental releases. 

Also almost all the releases should happen from a branch (say `master` or `branch-0.2`) and shouldn't require cherry-picking. We might cherrypick before as a part of backporting things.